packages/braid-design-system/src/lib/components/TooltipRenderer/TooltipRenderer.tsx

```diff
@@ -134,12 +156,21 @@ export const TooltipRenderer = ({
     setTriggerRef,
     triggerRef,
     getArrowProps,
+    update,
   } = usePopperTooltip(
     {
       placement,
       trigger: [isMobile() ? 'click' : 'hover', 'focus'],
       visible: isStatic || controlledVisible,
-      onVisibleChange: setControlledVisible,
+      onVisibleChange: (newState) => {
+        setControlledVisible(newState);
```

Comment:
Should we move this to the end of the callback, after the refs are updated? IIRC under some circumstances `set*State` can trigger a rerender.

Reply:
Can move it to end of the callback.